core/src/main/java/tc/oc/pgm/api/filter/query/MatchQuery.java

```diff
@@ -12,4 +16,16 @@ default Filterable<?> extractFilterable() {
     if (this instanceof PartyQuery) return ((PartyQuery) this).getParty();
     return this.getMatch();
   }
+
+  default <T extends MatchModule> T moduleRequire(Class<T> cls) {
+    return getMatch().needModule(cls);
+  }
+
+  default <T extends MatchModule> Optional<T> moduleOptional(Class<T> cls) {
+    return Optional.ofNullable(getMatch().getModule(cls));
+  }
```

Comment:
it's convenient

Reply:
Oh right this will be accessible through the actual match object as well, I thought it was weird having separate behaviors for the query and the match but it makes sense now.